test/utils.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-env node */
+// The geckodriver package downloads and installs geckodriver for us.
+// We use it by requiring it.
+
+require("geckodriver");
+const cmd = require("selenium-webdriver/lib/command");
+const firefox = require("selenium-webdriver/firefox");
+const Fs = require("fs-promise");
+const FxRunnerUtils = require("fx-runner/lib/utils");
+const path = require("path");
+const webdriver = require("selenium-webdriver");
+
+const By = webdriver.By;
+const Context = firefox.Context;
+const until = webdriver.until;
+
+// Note: Geckodriver already has quite a good set of default preferences
+// for disabling various items.
+// https://github.com/mozilla/geckodriver/blob/master/src/marionette.rs
+const FIREFOX_PREFERENCES = {
+  // Ensure e10s is turned on.
+  "browser.tabs.remote.autostart": true,
+  "browser.tabs.remote.autostart.1": true,
+  "browser.tabs.remote.autostart.2": true,
+  // These are good to have set up if you're debugging tests with the browser
+  // toolbox.
+  "devtools.chrome.enabled": true,
+  "devtools.debugger.remote-enabled": true,
+};
+
+// useful if we need to test on a specific version of Firefox
+async function promiseActualBinary(binary) {
+  try {
+    const normalizedBinary = await FxRunnerUtils.normalizeBinary(binary);
+    await Fs.stat(normalizedBinary);
+    return normalizedBinary;
+  } catch (ex) {
+    if (ex.code === "ENOENT") {
+      throw new Error(`Could not find ${binary}`);
+    }
+    throw ex;
+  }
+}
+
+module.exports.promiseSetupDriver = async() => {
+  try {
+    const profile = new firefox.Profile();
+
+    Object.keys(FIREFOX_PREFERENCES).forEach((key) => {
+      profile.setPreference(key, FIREFOX_PREFERENCES[key]);
+    });
+
+    const options = new firefox.Options();
+    options.setProfile(profile);
+
+    const builder = new webdriver.Builder()
+      .forBrowser("firefox")
+      .setFirefoxOptions(options);
+
+    const binaryLocation = await promiseActualBinary(process.env.FIREFOX_BINARY || "firefox");
+    await options.setBinary(new firefox.Binary(binaryLocation));
+    const driver = await builder.build();
+    driver.setContext(Context.CHROME);
+    return driver;
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+module.exports.addShareButton = async driver =>
+  driver.executeAsyncScript((callback) => {
+    // see https://dxr.mozilla.org/mozilla-central/source/browser/base/content/browser-social.js#193
```

Comment:
DXR links should probably be permalinks, so they don't break as the code changes overtime.